### PR TITLE
Remove NVTX.

### PR DESCRIPTION
**Pull request by maleadt**
The NVTX dynamic library shipped as part of CUDA only provides API v1, and CUDA 12 doesn't ship it at all (for Windows, at least). NVIDIA suggested building it from https://github.com/NVIDIA/NVTX#other-languages, which is apparently what @simonbyrne has done already for NVTX.jl. So let's just remove NVTX from CUDA.jl altogether.

cc @simonbyrne, maybe there's something to salvage here (e.g., the Clang.jl-generated headers?)

### CODE REVIEW

File: lib/cudadrv/state.jl [hunk 1]
```diff
@@ -356,10 +356,11 @@ end
...
     # register the name of this task
-    t = current_task()
-    tptr = pointer_from_objref(current_task())
-    tptrstr = string(convert(UInt, tptr), base=16, pad=Sys.WORD_SIZE>>2)
-    NVTX.nvtxNameCuStreamA(stream, "Task(0x$tptrstr)")
+    # XXX: do this when the user has imported NVTX.jl (using weak dependencies?)
+    #t = current_task()
+    #tptr = pointer_from_objref(current_task())
+    #tptrstr = string(convert(UInt, tptr), base=16, pad=Sys.WORD_SIZE>>2)
+    #NVTX.nvtxNameCuStreamA(stream, "Task(0x$tptrstr)")
```
Comment (simonbyrne) [inline lib/cudadrv/state.jl:363]: Would it make more sense to add these to NVTX.jl, as a WeakDep on CUDA?

Comment (maleadt) [reply lib/cudadrv/state.jl:363]: Yeah, but we're currently supporting down to Julia 1.6, hence just the comment. Or is there a backwards-compatible option for that?

**Comment by simonbyrne** [comment]
@maleadt Should I move NVTX.jl into JuliaGPU?

**Comment by maleadt** [comment]
Yeah sure, if you want to.

**Comment by simonbyrne** [comment]
I don't have sufficient permissions

File: lib/cudadrv/state.jl (hunk 1 above)
Comment (simonbyrne) [reply lib/cudadrv/state.jl:363]: I think we can have a fallback via Requires

**Comment by maleadt** [comment]
Can you try again?

**Comment by simonbyrne** [comment]
done, thanks!

**Comment by maleadt** [comment]
Great, thanks! I guess we'll need to update the URL in the General registry as well?

**Pull request merged**